src/bin/idl/mod.rs

```diff
@@ -367,6 +367,9 @@ fn idltype_to_solidity(ty: &IdlType, ty_names: &[(String, String)]) -> Result<St
             Ok(ty) => Ok(format!("{ty}[{size}]")),
             Err(ty) => Err(format!("{ty}[{size}]")),
         },
+        IdlType::Generic(..)
+        | IdlType::GenericLenArray(..)
+        | IdlType::DefinedWithTypeArgs { .. } => Err("generics are not supported".into()),
```

Comment:
I'm hoping you are prepared to overlook this warning (note that overall code coverage has increased)

Reply:
I liked this feature of pointing out lines not covered by tests.